packages/tasks/src/local-apps.ts

```diff
@@ -0,0 +1,119 @@
+import type { ModelData } from "./model-data";
+import type { PipelineType } from "./pipelines";
+
+/**
+ * Elements configurable by a local app.
+ */
+export type LocalApp = {
+	/**
+	 * Name that appears in buttons
+	 */
+	prettyLabel: string;
+	/**
+	 * Link to get more info about a local app (website etc)
+	 */
+	docsUrl: string;
+	/**
+	 * main category of app
+	 */
+	mainTask: PipelineType;
+	/**
+	 * Whether to display a pill "macOS-only"
+	 */
+	macOSOnly?: boolean;
+
+	comingSoon?: boolean;
+	/**
+	 * IMPORTANT: function to figure out whether to display the button on a model page's main "Use this model" dropdown.
+	 */
+	displayOnModelPage: (model: ModelData) => boolean;
+} & (
+	| {
+			/**
+			 * If the app supports deeplink, URL to open.
+			 */
+			deeplink: (model: ModelData) => URL;
+	  }
+	| {
+			/**
+			 * And if not (mostly llama.cpp), snippet to copy/paste in your terminal
+			 */
+			snippet: (model: ModelData) => string;
+	  }
+);
+
+function isGgufModel(model: ModelData) {
+	return model.tags.includes("gguf");
+}
+
+const snippetLlamacpp = (model: ModelData): string => {
+	return `llama-cpp \
```

Comment:
This doesn't work atm. :/

```
./main
```

Reply:
arg :(

care to suggest the replacement?

Reply:
Probably users can take the pre-built binary from here: https://github.com/ggerganov/llama.cpp/releases

Then use either `main` or `server`. Personally I recommend `server`, since `main` does not have support for chat templates.

The command will be `./server -m model.gguf` (remember to add `.exe` for windows). Then, a basic frontend then can be access via `http://localhost:8080`

Reply:
Also a small note: we also need to specify `--hf-file` along with `--hf-repo`, because gguf files does not have a fixed file name. For example:

```
./server --hf-repo ggml-org/models --hf-file tinyllamas/stories260K.gguf -m ./model.gguf
```

Though, that seems a bit complicated IMO, so I prefer to manually download the file and use `-m` to load it.

Reply:
```suggestion
	return `./main \
```